pkgs/tools/security/cosign/default.nix

```diff
@@ -2,29 +2,48 @@
 
 buildGoModule rec {
   pname = "cosign";
-  version = "1.5.2";
+  version = "1.6.0";
 
   src = fetchFromGitHub {
     owner = "sigstore";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-37jahAGgQn7HwwdRTlAS/oJQ3BxTkMViI6iJMBYFgjI=";
+    sha256 = "sha256-jAkTIO+tmb1vjS2eRWU9Fau7qzPCBlXJCk00iwNpULE=";
   };
 
   buildInputs = lib.optional (stdenv.isLinux && pivKeySupport) (lib.getDev pcsclite)
     ++ lib.optionals (stdenv.isDarwin && pivKeySupport) [ PCSC ];
 
   nativeBuildInputs = [ pkg-config installShellFiles ];
 
-  vendorSha256 = "sha256-d3aOX4iMlhlxgYbqCHCIFKXunVha0Fw4ZBmy4OA6EhI=";
+  vendorSha256 = "sha256-E9zeRlPIIoXo/EfagHC3aDnW747SdsPiqIA384D7NQI=";
 
-  excludedPackages = "\\(sample\\|webhook\\|help\\)";
+  subPackages = [
+    "cmd/cosign"
+    "cmd/cosign/webhook"
+    "cmd/sget"
+  ];
 
   tags = [] ++ lib.optionals pivKeySupport [ "pivkey" ] ++ lib.optionals pkcs11Support [ "pkcs11key" ];
 
-  ldflags = [ "-s" "-w" "-X github.com/sigstore/cosign/pkg/version.GitVersion=v${version}" ];
+  ldflags = [
+    "-s"
+    "-w"
+    "-X sigs.k8s.io/release-utils/version.gitVersion=v${version}"
+    "-X sigs.k8s.io/release-utils/version.gitTreeState=clean"
+  ];
+
+  postBuild = ''
+    # cmd/cosign/webhook should be called cosigned
+    mv $GOPATH/bin/{webhook,cosigned}
+  '';
+
+  # tests currently hang part way through
```

Comment:
This can be resolved by updating the list of removed files to
```
    rm pkg/cosign/tuf/client_test.go # Require network access
    rm cmd/cosign/cli/fulcio/fulcioroots/fulcioroots_test.go # Require network access
    rm pkg/cosign/tlog_test.go # Require network access
```

Reply:
Thanks. adding

```
    rm cmd/cosign/cli/fulcio/fulcioroots/fulcioroots_test.go # Require network access
    rm pkg/cosign/tlog_test.go # Require network access
```

to the existing list worked. Just using those 3 it got stuck (but a lot further in)

How did you go about debugging it?

Reply:
Yup sorry, I did not test with `cosigned`. The minimal list is 
```
    rm cmd/cosign/cli/fulcio/fulcioroots/fulcioroots_test.go # Require network access
    rm pkg/cosign/kubernetes/webhook/validator_test.go # Require network access
    rm pkg/cosign/tlog_test.go # Require network access
    rm pkg/cosign/tuf/client_test.go # Require network access
```

The two signer_test.go files succeed now (or I disabled them incorrectly on previous updates :/ ).

> How did you go about debugging it?

Go has a test timeout of 10 minutes so after some time you will get an helpful error message telling you which test block. To debug quickly I temporarily edit the checkPhase in pkgs/development/go-modules/generic/default.nix to something like this:
```
    checkPhase = args.checkPhase or ''
      runHook preCheck

      for pkg in $(getGoDirs test); do
        buildGoDir 'test -timeout=10s' $checkFlags "$pkg"
      done

      runHook postCheck
    '';
```

Reply:
ah ok, that's super helpful